apps/src/blockTooltips/DropletTooltipManager.js

```diff
@@ -67,14 +74,18 @@ DropletTooltipManager.prototype.registerDropletTextModeHandlers = function (drop
 };
 
 /**
- * @param {DropletBlock[]} dropletBlocks list of Droplet block definitions for
- *    which to register documentation
+ * Registers blocks based on the dropletBlocks and blockFilter passed to the constructor
  */
-DropletTooltipManager.prototype.registerBlocksFromList = function (dropletBlocks) {
-  dropletBlocks.forEach(function (dropletBlockDefinition) {
-    this.blockTypeToTooltip[dropletBlockDefinition.func] =
-      new DropletFunctionTooltip(this.appMsg, dropletBlockDefinition);
-  }, this);
+DropletTooltipManager.prototype.registerBlocks = function () {
+  dropletUtils.getAllAvailableDropletBlocks(this.dropletConfig).forEach(
+    function (dropletBlockDefinition) {
+      if (!dropletBlockDefinition.noAutocomplete &&
+          (!this.blockFilter || typeof this.blockFilter[dropletBlockDefinition.func] !== 'undefined')) {
+        this.blockTypeToTooltip[dropletBlockDefinition.func] =
+          new DropletFunctionTooltip(this.appMsg, dropletBlockDefinition);
+      }
+    },
```

Comment:
This is also getting very nested/difficult to parse. I would write it as:
```
function (dropletBlockDefinition) {
  if (dropletBlockDefinition.noAutoComplete) {
    return;
  }
  if (this.blockFilter && this.blockFilter[dropletBlockDefinition.func] !== undefined) {
    this.blockTypeToTooltip[dropletBlockDefinition.func] =
      new DropletFunctionTooltip(this.appMsg, dropletBlockDefinition);
  }
}
```
(not 100% sure i got all the conditionals right)

Alternatively, I wonder if it would make sense to have some sort of helper `function showTooltipForBlock` since we seem to have this logic in a couple places